`Development`: Add atomic button and molecule of button group

Douglas-Sarah opened the pull request:
<!-- Thanks for contributing to TumApply! Before you submit your pull request, please make sure to check all tasks by putting an x in the [ ] (don't: [x ], [ x], do: [x]). Remove not applicable tasks and do not leave them unchecked -->
<!-- If your pull request is not ready for review yet, create a draft pull request! -->

### Checklist

#### General

<!-- Remove tasks that are not applicable for your PR. Please only put the PR into ready for review, if all relevant tasks are checked! -->
<!-- You only need to choose one of the first two check items: Generally, test on the test servers. -->
<!-- If it's only a small change, testing it locally is acceptable, and you may remove the first checkmark. If you are unsure, please test on the test servers. -->

- [x] I tested **all** changes and their related features with **all** corresponding user types.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://confluence.aet.cit.tum.de/spaces/AP/pages/257786006/Language+Guidelines+Client).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://confluence.aet.cit.tum.de/spaces/AP/pages/256311714/PR+Guidelines).

#### Client

- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data (e.g. using paging).
- [x] I **strictly** followed the principle of **data economy** for all client-server REST calls.
- [x] I **strictly** followed the [client coding and design guidelines](https://confluence.aet.cit.tum.de/spaces/AP/pages/256311716/Client+Guidelines).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added screenshots/screencasts of my UI changes.

### Motivation and Context

Under atomic design we will implement basic components modularly to ensure uniformity among the whole application.

### Description

<!-- Describe your changes in detail -->
This PR implements the buttons and buttons groups of the basic components.

### Steps for Testing

<!-- Please describe in detail how reviewers can test your changes. Make sure to take all related features and views into account! Below is an example that you can refine. -->

The button types are visible under the path `/button`.

### Review Progress

<!-- Each PR should be reviewed by at least one other developers. The code, the functionality (= manual test). -->
<!-- The reviewer or author check the following boxes depending on what was reviewed or tested. All boxes should be checked before merge. -->
<!-- You can add additional checkboxes if it makes sense to only review parts of the code or functionality. -->
<!-- When changes are pushed, uncheck the affected boxes. (Not all changes require full re-reviews.) -->

#### Code Review

- [x] Code Review 1

#### Manual Tests

- [x] Test 1

### Screenshots
<!-- ![image](https://github.com/user-attachments/assets/4ea1296d-ea39-4d23-9bce-dc8369bb80c8) -->
![image](https://github.com/user-attachments/assets/14215ed8-4eff-490d-912b-919e65927920)

This page is visible under the path `/button` (e.g. locally under `localhost:4200/button`)
<!-- Add screenshots to demonstrate the changes in the UI. Remove the section if you did not change the UI. -->
<!-- Create a GIF file from a screen recording in a docker container https://toub.es/2017/09/11/high-quality-gif-with-ffmpeg-and-docker/ -->

Douglas-Sarah commented:
I used the old scss color variables for styling instead of integrating the new theme completely - I will integrate the theme in the next week

Douglas-Sarah commented:
It is by no means perfect, but I fixed the circular button issue, and my implementation of the disabled button (there is only on color for disabled buttons now, instead of setting them transparent)
They now also use the correct color, but there is still an issue with using the scss variables at places

az108 commented on src/main/webapp/app/shared/components/atoms/button/button.component.scss:
```diff
@@ -0,0 +1,176 @@
+  padding: 0.3em;
+  position: absolute;
+  min-width: 1.2rem;
+  right: -0.6em;
```
can we avoid negative values here or is it not possible at all?

Douglas-Sarah replied:
removed in newest commit

az108 commented on src/main/webapp/app/shared/components/atoms/button/button.component.scss:
```diff
@@ -0,0 +1,176 @@
+.info_text {
+  @include text-button(black);
+}
+.badge_mini {
```
do we need all of stylings?

Douglas-Sarah replied:
At the moment, yes, as the new theme for the buttons is not complete yet. This is also the reason for all the !importants. I will solve that with Marc tomorrow

Douglas-Sarah replied:
I have removed the majority of the scss code

az108 commented on src/main/webapp/app/shared/components/atoms/button/button.component.scss:
```diff
@@ -0,0 +1,176 @@
+
+  &.disabled,
+  &:disabled {
+    background: lighten($base-color, 40%) !important;
```
do we need so many !importants . Ideally we avoid this and only use it when absolutely necessary when we have to override some prime ng stuff.

Douglas-Sarah replied:
At the moment the !importants override the primeng stuff- but tomorrow I will extend the theme, so the styling should reduce significantly

Douglas-Sarah replied:
Overriding is not necessary anymore

az108 commented on src/main/webapp/app/shared/components/atoms/button/button.component.scss:
```diff
@@ -0,0 +1,176 @@
+
+@mixin outlined-button($color) {
+  background: white !important;
+  border: 0.06em solid $color !important;
```
do we need such a em value. First we need rem.

Douglas-Sarah replied:
Changed to rem with the newest commit

az108 commented on src/main/webapp/app/shared/components/atoms/button/button.component.scss:
```diff
@@ -0,0 +1,176 @@
+  @include text-button($negative-default);
+}
+
+// .accent_outlined {
```
remove this unneccessary code

Douglas-Sarah replied:
Removed 👍

az108 commented on src/main/webapp/app/shared/components/atoms/button/button.component.ts:
```diff
@@ -1,12 +1,89 @@
+ */
+export type ButtonVariant = 'filled' | 'outlined' | 'text';
+
+/**
```
i would remove all comments like this where its clear what the code is doing

az108 commented on src/main/webapp/app/shared/components/atoms/button/button.component.ts:
```diff
@@ -1,12 +1,89 @@
+   * @default 'filled'
+   */
+  variant = input<ButtonVariant>('filled');
+
```
same here, i dont thing we need the comments.

Douglas-Sarah replied:
Removed 👍

az108 commented on src/main/webapp/app/shared/components/molecules/button-group/button-group.component.scss:
```diff
@@ -0,0 +1,9 @@
+jhi-button {
+  margin-bottom: 0.25em;
```
rem please

Douglas-Sarah replied:
Changed to rem with the newest commit

az108 commented on src/main/webapp/app/shared/components/molecules/button-group/button-group.component.ts:
```diff
@@ -0,0 +1,40 @@
+  buttons: Button[];
+};
+
+/**
```
this is the only comment i would leave here

az108 commented on src/main/webapp/app/shared/components/molecules/button-group/button-group.component.ts:
```diff
@@ -0,0 +1,40 @@
+/**
+ * Defines the data structure for a group of buttons.
+ *
+ * @property direction - Layout direction of the button group:
```
and this comment is fine aswell

az108 commented on src/main/webapp/content/theming/tumapplypreset.ts:
```diff
@@ -299,8 +299,8 @@ export const TUMApplyPreset = definePreset(Lara, {
       root: {
-        borderRadius: '16px',
-        fontWeight: '500',
+        borderRadius: '8px',
```
avoid px please

Douglas-Sarah replied:
Changed to rem 👍

az108 reviewed (changes requested):
Code LGTM overall, except two things.

Please just use comments on client side if the code is complicated or not understandable. And use it for explaining different components.

Second important point would be that you need to check all the css stylings if they are really neccessary. I think we could remove many of them and maybe even get rid of them. You can check with ishani as she had minimalistic css for her atomic components.

ani-ib commented on src/main/webapp/app/shared/components/atoms/button/button.component.scss:
```diff
@@ -0,0 +1,157 @@
+  color: $text-color !important;
+
+  &:hover {
+    background: darken($neutral-300, 10%) !important;
```
Each button type (e.g. primary, secondary, positive, etc.) has a specific hover color that reflects its role. It might make sense for the filled buttons to use that corresponding hover color as their background when hovered—for example, a primary button would use the primary-hover background.

No pressure to update it right now, but I wanted to point it out in case it fits into your styling changes tomorrow.

Douglas-Sarah replied:
This is all solved through the primeng theme now

ani-ib commented on src/main/webapp/app/shared/components/atoms/button/button.component.scss:
```diff
@@ -0,0 +1,157 @@
+  }
+}
+
+@mixin filled-button($color, $text-color: black, $text-color-disabled: black) {
```
Why do we need "$text-color-disabled"? I would suggest removing it if it is not being used anywhere.
```suggestion
@mixin filled-button($color, $text-color: black) {
```

Douglas-Sarah replied:
This is all solved through the primeng theme now

ani-ib commented on src/main/webapp/app/shared/components/atoms/button/button.component.scss:
```diff
@@ -0,0 +1,157 @@
+  @include outlined-button($primary-default);
+}
+.primary_filled {
+  @include filled-button($primary-default, white, white);
```
If you remove "$text-color-disabled" in the other comment then please remove it from here as well
```suggestion
  @include filled-button($primary-default, white);
```

Douglas-Sarah replied:
This is all solved through the primeng theme now

ani-ib commented on src/main/webapp/content/theming/tumapplypreset.ts:
```diff
@@ -336,7 +336,7 @@ export const TUMApplyPreset = definePreset(Lara, {
             disabledText: '{text.disabled}',
           },
-          warning: {
+          warn: {
```
Was there a particular purpose to adding these changes into this PR?

ani-ib reviewed (changes requested):
Code looks good overall - added minor comments :)
Note: Don't forget to remove values such as "white" and hardcoded colors when updating the changes in the next PR

ani-ib commented on src/main/webapp/app/shared/components/atoms/button/button.component.html:
```diff
@@ -1 +1,20 @@
+    [variant]="variant()"
+    [severity]="color()"
+    [raised]="numberOfFavorites() !== undefined"
+    badgeStyle=""
```
Please remove the badgeStyle if it is unnecessary

ani-ib reviewed (changes requested):
I have tested the button page locally and noticed the following style issues:
<img width="270" alt="image" src="https://github.com/user-attachments/assets/fd148328-1592-4309-9e11-87932c43e164" />
In the above image, when hovering on the secondary filled button, the text color is too dark and not legible.
<img width="270" alt="image" src="https://github.com/user-attachments/assets/b543f58c-a6b9-4f58-9cb3-2721c1089120" />
In the above image, when hovering on the warn button, the border looks red.
<img width="540" alt="image" src="https://github.com/user-attachments/assets/3659a779-707e-4a15-aa9d-4e4a8ac265b6" />
In the above image, when hovering on the neutral button, the border is blue instead of neutral.
<img width="540" alt="image" src="https://github.com/user-attachments/assets/b4e8af8a-036a-45ec-82ad-6781f8a34ae2" />

<img width="540" alt="image" src="https://github.com/user-attachments/assets/b4e8af8a-036a-45ec-82ad-6781f8a34ae2" />
The text-color in the non-filled secondary buttons doesn't match the background color of the filled secondary buttons.

marc-fett commented on src/main/webapp/app/shared/components/molecules/button-group/button-group.component.scss:
```diff
@@ -0,0 +1,9 @@
+  margin-right: 0.25rem;
+}
+
+.button_group {
```
please use - instead of _ in scss and maybe include it in the div directly instead of a new scss class

Douglas-Sarah replied:
Adjusted in latest commits 👍

marc-fett commented on src/main/webapp/app/shared/components/molecules/button-group/button-group.component.spec.ts:
```diff
@@ -0,0 +1,46 @@
+  it('should render the correct number of buttons', () => {
+    const buttonElements = fixture.nativeElement.querySelectorAll('jhi-button');
+    expect(buttonElements.length).toBe(mockButtonGroupData.buttons.length);
+  });
```
Please add a test if the buttons are clickable e.g.

``
it('should call onClick when a button is clicked', () => {
  const spy = spyOn(mockButtonGroupData.buttons[0], 'onClick');
  const buttonElement = fixture.nativeElement.querySelector('jhi-button');
  buttonElement.click();
  expect(spy).toHaveBeenCalled();
});
``

Douglas-Sarah replied:
Adjusted in latest commits 👍

Douglas-Sarah replied:
Nice test btw

marc-fett commented on src/main/webapp/app/shared/components/molecules/button-group/button-group.component.ts:
```diff
@@ -0,0 +1,32 @@
+ * @property buttons - Array of `Button` objects to render.
+ */
+export type ButtonGroupData = {
+  direction: 'vertical' | 'horizontal';
```
would add a type for this
export type ButtonGroupDirection = 'vertical' | 'horizontal';

Douglas-Sarah replied:
Adjusted in latest commits 👍

marc-fett commented on src/main/webapp/app/shared/components/atoms/button/button.component.ts:
```diff
@@ -1,12 +1,36 @@
+};
+
 @Component({
   selector: 'jhi-button',
```
I would name the component a bit different to distinguish between Jhipster generated components and our components. E.g. custom-button or app-button

Douglas-Sarah replied:
Resonable suggestion.
This is the suggested name if we generate the component via `ng generate component` - it is because of this line in the angular.json:
![image](https://github.com/user-attachments/assets/11fdf8e5-c5a9-44d7-a916-a84bedc52a78)

I would suggest renaming all our jhi components within another commit- all our new components are named that way at the moment.

Do you know if jhipster also needs the prefix - variable in the angular.json?

marc-fett replied:
afaik jhipster doesn't need the prefix, so we can change the prefix to something else and all future components will automatically have the right name.

But I agree, it is better to do it in another PR.

marc-fett commented on src/main/webapp/app/shared/components/atoms/button/button.component.html:
```diff
@@ -1 +1,20 @@
+    [label]="label() ?? ''"
+  >
+    @if (icon() !== undefined) {
+      <fa-icon [icon]="['fas', icon() ?? '']" />
```
icon() can't be undefined anymore so the second check is not necessary

Douglas-Sarah replied:
![image](https://github.com/user-attachments/assets/f397681e-4c95-4b19-b76f-5a83fc4d9e63)
![image](https://github.com/user-attachments/assets/cf07b678-6a18-4255-9e6b-7e4b78be3a34)

I guess it does not work like that because the icon() fetches the value each time and that could be a different value in the eyes of the compiler. I am not knowledgeable enough with signals to know if this is true. The `@if` is needed anyway, because i do not want to render the icon if there is no icon to render.

If you have a more elegant solution, I am open for suggestions

marc-fett replied:
Ah yes forgot about that. Then everything is fine :)

marc-fett commented on src/main/webapp/app/shared/components/molecules/button-group/button-group.component.html:
```diff
@@ -0,0 +1,15 @@
+<div class="button_group flex {{ data().direction === 'vertical' ? 'flex-col' : 'flex-row' }}">
+  @for (button of data().buttons; track $index) {
+    <jhi-button
+      class="'grow-1'"
```
please remove the ' ':

class="grow-1"
[buttonClass]="w-full"

marc-fett commented on src/main/webapp/app/shared/components/atoms/button/button.component.ts:
```diff
@@ -1,12 +1,36 @@
+  onClick: VoidFunction;
+};
+
 @Component({
```
missing tests for the button component. There are only tests for the button group component.

Douglas-Sarah replied:
I added tests with the latest commit 👍

marc-fett commented on src/main/webapp/app/shared/components/atoms/button/button.component.scss:
```diff
@@ -0,0 +1,27 @@
+  color: var(--p-color-disabled);
+}
+
+.badge_mini {
```
please name it according to BEM convention e.g.

.jhi-button__badge--mini

- jhi-button => component name
- __badge => element within the component
- --mini => modifier

(or different start name if we want to rename jhi-button to something else)

Douglas-Sarah replied:
Adjusted in latest commit 👍

Douglas-Sarah replied:
Somehow that format did not work with the angular compiler- changed it to `.jhi-button-badge-mini`.
Feel free to commit the working version of this css class suggestion directly to the branch.

marc-fett reviewed (changes requested):
I can see the same problems, Ishani found. Some Buttons have strange hover and click colors and borders.

You also often use double quotes like "'.....'" in the code. Please search for occurrences and replace it with "..."

Douglas-Sarah commented:
> I can see the same problems, Ishani found. Some Buttons have strange hover and click colors and borders.
> 
> You also often use double quotes like "'.....'" in the code. Please search for occurrences and replace it with "..."

Thank you for your suggestions. Does the singe quote issue also pertain to the generated code?
![image](https://github.com/user-attachments/assets/9ffc10cd-8538-44d4-ae03-70fb27e89bf2)

The generated code here also uses single quotes - I tried to be consistent.

Douglas-Sarah commented:
I fixed the border and the hover issue and cleaned up the theme:
![image](https://github.com/user-attachments/assets/49dd8fde-ebe4-4a11-b390-ad9ffae50e07)
Hover:
![image](https://github.com/user-attachments/assets/785addf0-3d76-4fea-aa30-c14bf6d1ea8b)
Dark mode:
![image](https://github.com/user-attachments/assets/d4970785-d634-411e-8b61-8b3ae20a2356)
Dark mode hover:
![image](https://github.com/user-attachments/assets/9bce4266-87d5-4c45-b1e6-4f5d1debc7ec)

Douglas-Sarah commented:
> > I can see the same problems, Ishani found. Some Buttons have strange hover and click colors and borders.
> > You also often use double quotes like "'.....'" in the code. Please search for occurrences and replace it with "..."
> 
> Thank you for your suggestions. Does the singe quote issue also pertain to the generated code? ![image](https://private-user-images.githubusercontent.com/198627919/443416767-9ffc10cd-8538-44d4-ae03-70fb27e89bf2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDcxNzU1ODUsIm5iZiI6MTc0NzE3NTI4NSwicGF0aCI6Ii8xOTg2Mjc5MTkvNDQzNDE2NzY3LTlmZmMxMGNkLTg1MzgtNDRkNC1hZTAzLTcwZmIyN2U4OWJmMi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNTEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDUxM1QyMjI4MDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hNTg3NzdmNjgwZGQ4OWRmMmQ4YjEzZmNmZDhiNzg2MjI0N2IzN2QxNTZmMThlNzljZGY0NWMxNzMyYjQxNTMyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.oSB57bVoUhR213mlfkjEkFgiLfsnvh7yrINE3kGBnZE)
> 
> The generated code here also uses single quotes - I tried to be consistent.

I misunderstood your point- I thought you meant in the .ts files, but you meant in the .html files. (We talked off-github)
I adjusted that within one of the latest commits

ani-ib reviewed (changes requested):
I added some small comments
<img width="1263" alt="image" src="https://github.com/user-attachments/assets/03267d4f-d82d-4725-aa64-f89bb63f6ddf" />
<img width="935" alt="image" src="https://github.com/user-attachments/assets/9c3f828e-4013-4063-8a68-4f6e12b1c59b" />
For filled buttons, can you make the border color the same as the background color? This would adhere to the Figma design and would look cleaner

Douglas-Sarah commented on src/main/webapp/app/shared/components/atoms/button/button.component.html:
```diff
@@ -1 +1,19 @@
+  <p-button
+    [disabled]="disabled()"
+    [variant]="variant()"
+    [severity]="color()"
```
It is called color in the Figma design:
![image](https://github.com/user-attachments/assets/5efb3036-064a-4604-855c-c59200d8b77e)
Do you think I still need to change it?

marc-fett replied:
I can change the naming in Figma. It is a good suggestion. Just didn't know it was called severity in PrimeNG.

Douglas-Sarah commented:
> I added some small comments <img alt="image" width="1263" src="https://private-user-images.githubusercontent.com/88331280/443592896-03267d4f-d82d-4725-aa64-f89bb63f6ddf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDcyMTczMTYsIm5iZiI6MTc0NzIxNzAxNiwicGF0aCI6Ii84ODMzMTI4MC80NDM1OTI4OTYtMDMyNjdkNGYtZDgyZC00NzI1LWFhNjQtZjg5YmI2M2Y2ZGRmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA1MTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNTE0VDEwMDMzNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE4MjAyNGI1ZmM0NTYyMDQ0NzZmZjZmOTI5NWRhZDI3ZmZmNDFmMTE3ZjEwZDhjZDY0ZGM1MTEwOTUwZGMxMDkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.3EtouhmkGp2TXoTvAxjjX7_PGGLHYR4SEYpHvi50J9k"> <img alt="image" width="935" src="https://private-user-images.githubusercontent.com/88331280/443593914-9c3f828e-4013-4063-8a68-4f6e12b1c59b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDcyMTczMTYsIm5iZiI6MTc0NzIxNzAxNiwicGF0aCI6Ii84ODMzMTI4MC80NDM1OTM5MTQtOWMzZjgyOGUtNDAxMy00MDYzLThhNjgtNGY2ZTEyYjFjNTliLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA1MTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNTE0VDEwMDMzNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFhMjk1M2YwODc0MmI1MTk5OTUxOTMyNmY0ZThiYmYxMGZhOGUyNmNiZWUxMTg4NmUwOTQ1NWI2Y2IzOTYwNmEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.1iZQeu_JZFe5YC75T_6DUurFpkX9eFlMr_8KlhEBZ3M"> For filled buttons, can you make the border color the same as the background color? This would adhere to the Figma design and would look cleaner

Fixed in the latest commit:
![image](https://github.com/user-attachments/assets/7e9b481d-aeaa-4cac-9263-22666b3b5634)

Douglas-Sarah commented on src/main/webapp/app/shared/components/atoms/button/button.component.scss:
```diff
@@ -0,0 +1,27 @@
+
+::ng-deep p-button > .p-button-outlined:disabled {
+  background: var(--p-background-default);
+  border: 0.06rem var(--p-background-disabled) solid;
```
Fixed in the latest commit

ani-ib replied:
Thanks :)

Douglas-Sarah commented on src/main/webapp/app/shared/components/atoms/button/button.component.scss:
```diff
@@ -0,0 +1,27 @@
+  color: var(--p-color-disabled);
+}
+
+::ng-deep p-button > .p-button-text:disabled {
```
Fixed in the latest commit

Douglas-Sarah commented on src/main/webapp/app/shared/components/atoms/button/button.component.scss:
```diff
@@ -0,0 +1,27 @@
+  color: var(--p-color-disabled);
+}
+
+::ng-deep p-button > .p-button-outlined:disabled {
```
Fixed in the latest commit

Douglas-Sarah commented on src/main/webapp/app/shared/components/atoms/button/button.component.scss:
```diff
@@ -0,0 +1,27 @@
+::ng-deep p-button > button:disabled {
```
Fixed in the latest commit

ani-ib replied:
Thank you

marc-fett commented on src/main/webapp/app/shared/components/atoms/button/button.component.ts:
```diff
@@ -1,12 +1,37 @@
+  label = input<string | undefined>(undefined);
+  numberOfFavorites = input<number | undefined>(undefined);
+  disabled = input<boolean>(false);
+  buttonClass = input<string>('');
```
this button class is not used in the button component

Douglas-Sarah replied:
removed in latest commit

Douglas-Sarah commented on src/main/webapp/app/pages/misc/button-play-ground/button-play-ground.component.html:
```diff
@@ -0,0 +1,75 @@
+<div class="playground m-4">
+  <div class="row">
+    <jhi-button color="primary" [variant]="undefined" label="Submit" icon="sync" />
```
We talked about it off-github. Filled / undefined is already the standard.
But I will take it out of the button playground, so that others can copy a minimal version

Douglas-Sarah replied:
removed in latest commit

Douglas-Sarah commented on src/main/webapp/app/shared/components/atoms/button/button.component.html:
```diff
@@ -1 +1,19 @@
+    [label]="label()"
+  >
+    @if (icon() !== undefined) {
+      <fa-icon [icon]="['fas', icon() ?? '']" />
```
adjusted in latest commit

az108 commented:
Follow Up PR: 
check OnClick
Tooltip
Aria Label

az108 commented on src/main/webapp/app/shared/components/atoms/button/button.component.ts:
```diff
@@ -1,12 +1,37 @@
+  label?: string;
+  numberOfFavorites?: number;
+  disabled: boolean;
+  onClick: VoidFunction;
```
follow up

ani-ib reviewed (approved):
Code LGTM - thanks for the effort, the buttons look very well-designed and match the Figma designs well :)
